File: Task-03/Sub-Task 4/JavaScript/Ques_4.js
const fs = require('fs');

function readFromFile(filename) {
  try {
    return parseInt(fs.readFileSync(filename, 'utf8').trim());
  } catch (err) {
    console.error('Error reading from file:', err);
    return null;
  }
}

function generateDiamond(n) {
  let diamond = '';

  for (let i = 1; i <= n; i += 2) {
    for (let j = 0; j < (n - i) / 2; j++) {
      diamond += ' ';
    }
  
    for (let k = 0; k < i; k++) {
      diamond += '*';
    }

    diamond += '\n';
  }

  for (let i = n - 2; i > 0; i -= 2) {
    for (let j = 0; j < (n - i) / 2; j++) {
      diamond += ' ';
    }

    for (let k = 0; k < i; k++) {
      diamond += '*';
    }

    diamond += '\n';
  }

  return diamond.trim();
}

function writeToFile(filename, content) {
  try {
    fs.writeFileSync(filename, content);
    console.log('Diamond pattern written to file:', filename);
  } catch (err) {
    console.error('Error writing to file:', err);
  }
}

const n = readFromFile('input.txt');

if (n !== null) {
  const diamondPattern = generateDiamond(n);

  writeToFile('output.txt', diamondPattern);
}

File: Task-03/Sub-Task 4/Java/Ques_4.java
import java.io.*;

public class Ques3 {
    public static void main(String a[]) {
        try {
            int n = readFromFile("input.txt");
            generateDiamond(n);
        } catch (IOException e) {
            System.out.println("An error occurred: " + e.getMessage());
        }
    }

    public static int readFromFile(String filename) throws IOException {
        try (BufferedReader reader = new BufferedReader(new FileReader(filename))) {
            return Integer.parseInt(reader.readLine().trim());
        }
    }

    public static void generateDiamond(int n) {
        try (BufferedWriter writer = new BufferedWriter(new FileWriter("output.txt"))) {
          
            for (int i = 1; i <= n; i += 2) {
                for (int j = 0; j < (n - i) / 2; j++) {
                    writer.write(" ");
                }

                for (int k = 0; k < i; k++) {
                    writer.write("*");
                }
                writer.newLine();
            }

            for (int i = n - 2; i > 0; i -= 2) {
                for (int j = 0; j < (n - i) / 2; j++) {
                    writer.write(" ");
                }
                for (int k = 0; k < i; k++) {
                    writer.write("*");
                }

                writer.newLine();
            }
        } catch (IOException e) {
            System.out.println("An error occurred: " + e.getMessage());
        }
    }
}
